nucliadb_node/nucliadb_node/indexer.py

```diff
@@ -0,0 +1,337 @@
+
+        try:
+            await self._index_message(work.index_message)
+        except Exception as exc:
+            raise exc
+        else:
+            await work.nats_msg.ack()
+
+            await signals.successful_indexing.dispatch(
+                SuccessfulIndexingPayload(
+                    seqid=work.seqid, index_message=work.index_message
+                )
+            )
+            logger.info(
+                "Message indexing finished",
+                extra={
+                    "seqid": work.seqid,
+                    "shard": work.index_message.shard,
+                    "storage_key": work.index_message.storage_key,
+                    "time": time.time() - start,
+                },
+            )
+        finally:
+            await work.mpu.end()
+
+    async def _index_message(self, pb: IndexMessage):
+        status = None
+        if pb.typemessage == TypeMessage.CREATION:
+            status = await self._set_resource(pb)
+        elif pb.typemessage == TypeMessage.DELETION:
+            status = await self._delete_resource(pb)
+
+        if status is not None and status.status != OpStatus.Status.OK:
+            raise IndexNodeError(status.detail)
+
+    async def _set_resource(self, pb: IndexMessage) -> Optional[OpStatus]:
+        brain = await self.storage.get_indexing(pb)
+        shard_id = pb.shard
+        rid = brain.resource.uuid
+        brain.shard_id = brain.resource.shard_id = shard_id
+
+        logger.info(f"Adding {rid} at {shard_id} otx:{pb.txid}")
+        try:
+            status = await self.writer.set_resource(brain)
+
+        except AioRpcError as grpc_error:
+            if grpc_error.code() == StatusCode.NOT_FOUND:
+                logger.error(f"Shard does not exist {pb.shard}")
+            else:
+                event_id = errors.capture_exception(grpc_error)
+                logger.error(
+                    "An error ocurred on indexer worker while setting a resource. "
+                    f"Check sentry for more details. Event id: {event_id}"
+                )
+                if brain.HasField("metadata"):
+                    # Hard fail if we have the correct data
+                    raise grpc_error
+            return None
+
+        else:
+            logger.info(f"...done (Added {rid} at {shard_id} otx:{pb.txid})")
+            return status
+
+    async def _delete_resource(self, pb: IndexMessage) -> Optional[OpStatus]:
+        shard_id = pb.shard
+        rid = pb.resource
+        resource = ResourceID(uuid=rid, shard_id=shard_id)
+
+        logger.info(f"Deleting {rid} in {shard_id} otx:{pb.txid}")
+        try:
+            status = await self.writer.delete_resource(resource)
+
+        except AioRpcError as grpc_error:
+            if grpc_error.code() == StatusCode.NOT_FOUND:
+                logger.error(f"Shard does not exist {pb.shard}")
+            else:
+                event_id = errors.capture_exception(grpc_error)
+                logger.error(
```

Comment:
should this bubble  the error?

Reply:
We weren't bubbling it before, but if it makes sense I can do it